[GitHub] Fix slow sccache install on macOS by upgrading macOS version

linux4life798 opened the pull request:
The "Setup ccache" step on macOS-11 builds takes between 15 to 20 mins, whereas this step takes a less than 10 seconds on other runners. The bulk of this time is spent at the "Install sccache" step, where brew emits warnings like "Warning: You are using macOS 11." and "We (and Apple) do not provide support for this old version...".

Bumping the version of macOS greatly decreases this cache setup time to about 20 seconds. Furthermore, it seems like it is speeding up general build times, too.

It appears that https://github.com/actions/virtual-environments/issues/5900 has been resolved or obsoleted, so I do not believe
we need to lock macOS to 11 anymore.

linux4life798 committed:
[GitHub] Fix slow sccache install on macOS by upgrading macOS version

The "Setup ccache" step on macOS-11 builds takes between 15 to 20 mins,
whereas this step takes a less than 10 seconds on other runners.
The bulk of this time is spent at the "Install sccache" step, where brew
emits warnings like "Warning: You are using macOS 11." and
"We (and Apple) do not provide support for this old version...".

Bumping the version of macOS greatly decreases this cache setup time to
about 20 seconds. Furthermore, it seems like it is speeding up general
build times, too.

It appears that https://github.com/actions/virtual-environments/issues/5900
has been resolved, so I do not believe we need to lock macOS to 11 anymore.

linux4life798 commented:
@tstellar, is there any way to confirm that the issue mentioned above is resolved or obsolete?

tstellar commented on .github/workflows/llvm-project-tests.yml:
```diff
@@ -34,9 +34,7 @@ on:
-        # We're using a specific version of macOS due to:
-        # https://github.com/actions/virtual-environments/issues/5900
-        default: '["ubuntu-latest", "windows-2019", "macOS-11"]'
+        default: '["ubuntu-latest", "windows-2019", "macOS-latest"]'
```
We should use `macOS-12` rather than `macOS-latest` so we have control over when we update to the next version.

linux4life798 replied:
done.

tstellar commented:
> @tstellar, is there any way to confirm that the issue mentioned above is resolved or obsolete?

You can try removing the [workaround](https://github.com/llvm/llvm-project/blob/09e0d71ad34169e075e146e13a93b2d79f992e1d/.github/workflows/llvm-project-tests.yml#L95) and see if it still builds.

linux4life798 committed:
[GitHub] Remove pkg-config workaround for macOS-10.5

Remove the workaround for
https://github.com/actions/runner-images/issues/5900, which seems to
be resolved.

linux4life798 commented:
Test run with `macOS-latest`, but with the `PKG_CONFIG_PATH` env var removed.

https://github.com/linux4life798/llvm-project/actions/runs/7428652707/job/20216260090

linux4life798 commented:
It looks like be are building properly on macOS-12 without the `PKG_CONFIG_PATH` workaround.

Here is another test run using `macOS-12` from eb38995d6493623e5368d2b395d6ccc6c01bd612. The macOS-12 has finished building and succeeded.

* The enabled llvm projects are `clang;lldb;libclc;lld`
* The build target was ""

https://github.com/linux4life798/llvm-project/actions/runs/7428836371/job/20216662446

linux4life798 commented:
I forgot to change the `macOS-11` reference in the `workflow_dispatch` section. This PR is ready for review.

tstellar reviewed (approved):
LGTM, thanks.

linux4life798 commented:
@tstellar, I don't have merge rights to this repo. What is the protocol for merging approved changed?

tstellar commented:
@linux4life798 Which other PRs do you need merged?

linux4life798 commented:
* https://github.com/llvm/llvm-project/pull/76784
* https://github.com/llvm/llvm-project/pull/76914

linux4life798 commented:
:pray: thank you! The local commit stack was getting pretty deep.

linux4life798 commented:
It looks like there is some sccache / [hendrikmuhs/ccache-action]( https://github.com/hendrikmuhs/ccache-action) issue with macOS-12.

I explained the problem in https://github.com/hendrikmuhs/ccache-action/issues/174, but basically macOS-12 builds will appears to restore their cache correctly, but the build doesn't appear to be using the cache (really slow), and then it fails to get the cache stats at the end of the run. This last part causes the cache to not be saved.

I'm running an experiment with macOS-11 and macOS-12 to see if the version bump alone triggers this:

* [Build #58 with clean/deleted cache](https://github.com/linux4life798/llvm-project-ci-test/actions/runs/7435407194)
* [Build #63 using this newly generated cache](https://github.com/linux4life798/llvm-project-ci-test/actions/runs/7436439380)

linux4life798 commented:
Hmm. It looks like macOS-11 is does not see this cache issue, despite still having the `PKG_CONFIG_PATH` workaround removed.

I also tried to see if there was newer version of [hendrikmuhs/ccache-action](https://github.com/hendrikmuhs/ccache-action), but it looks like we are already using the newest version (the v1 tag was updated to the latest tag).

I am running another experiment to see if an older version of [hendrikmuhs/ccache-action](https://github.com/hendrikmuhs/ccache-action) still shows this cache issue, but in the meantime, I think we should just revert the commit for this PR (602c8fa2d8da).
